rfcs/expected_fail_message_tag_for_subtests.md

```diff
@@ -0,0 +1,71 @@
+# RFC 165: Support expected-fail-message in test metadata for subtests
+
+## Summary
+
+Add "expected-fail-message" in test metadata so that we can track how
+a subtest fails.
+
+## Details
+
+In Chromium developers are using baselines to track expected output message
+for testharness tests. While the output message for a pass subtest is
+generally not interesting, the output message for a failed subtest tracks
+how a subtest fails, and a change in how a test fails could be unintended
+and suggestive of a problem. Chromium developers want to keep such capability
+if we want to use Wptrunner to run WPTs in Chromium.
+
+The suggestion is to support "expected-fail-message" tag for subtests. When
+present, it takes effect on expected FAIL or expected PRECONDITION_FAILED
+results. The test runner should check if the actual output message matches
+with the "expected-fail-message". When a mismatch is found, the test runner
+should turn the results to unexpected FAIL or unexpected PRECONDITION_FAILED
+respectively, otherwise the results continue to be the expected one.
+
+Wdspec tests can also have subtests. "expected-fail-message" should work the
+same way for an expected FAIL Wdspec subtest. Other than that, "expected-fail-message"
+should not have impact on any other scenarios.
+
+One example metadata file will look like below:
+```
+[event.html]
+  expected: OK
+  [Changing selection from handler: input event]
+    expected: FAIL
+
+  [Command createLink, value "": input event]
+    expected: [FAIL, PRECONDITION_FAILED, PASS]
+    expected-fail-message: "number of input events fired expectei 1 but got 0"
+```
+
+### Implementations
+
+The current implementation for test metadata parser is flexible enough to
+understand any additional tag. The changes needed is 1) wpttest.Test to have one
+additional member function to return "expected-fail-message",
+2)testrunner.TestRunnerManager.test_ended to have additional logic to turn
+expected FAIL or PRECONDITION_FAILED to unexpected failures per the rule
+discribed above.
+
+A lint rule is also needed to prevent dangling "expected-fail-message", i.e. a
```

Comment:
Do we lint ini files at all now? Or is this a lint that will have to be in Chromium?

Reply:
I included this for completeness. We do lint ini files in Chromium but not at upstream now. As we are going to use test expectations/baselines in Chromium, we won't lint this either in Chromium. Reworded this to "A lint rule might be needed ...".

https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/tools/blinkpy/tool/commands/lint_wpt.py;drc=8185de7bf7ac655feddf94e7db7d544d6abb3e5c;l=67